CodeMaid/Logic/Cleaning/FileHeaderLogic.cs

```diff
@@ -58,16 +61,72 @@ internal void UpdateFileHeader(TextDocument textDocument)
                 return;
             }
 
+            settingsFileHeader = ReplaceVariables(textDocument, settingsFileHeader);
+
             var cursor = textDocument.StartPoint.CreateEditPoint();
             var existingFileHeader = cursor.GetText(settingsFileHeader.Length);
 
-            if (!existingFileHeader.StartsWith(settingsFileHeader))
+            if (!existingFileHeader.StartsWith(textDocument.GetCodeLanguageComment()))
             {
                 cursor.Insert(settingsFileHeader);
                 cursor.Insert(Environment.NewLine);
             }
         }
 
+        /// <summary>
+        /// Replace the variables in the header if any
+        /// </summary>
+        /// <param name="textDocument"></param>
+        internal string ReplaceVariables(TextDocument textDocument, string settingsFileHeader)
+        {
+            var outputSettingsFileHeader = settingsFileHeader;
+
+            foreach (var variable in _variables)
+            {
+                // If the file header doesn't contain this variable, do nothing...
+                if (!settingsFileHeader.Contains(variable))
+                {
+                    continue;
+                }
+
+                var variableValue = string.Empty;
+                switch (variable)
+                {
+                    case "$SOLUTION$":
```

Comment:
Since these strings are repeated in a couple places, please make them constants.  The _variables array can be built from the constants.